src/js/server/lib/datasource.js

```diff
@@ -0,0 +1,160 @@
+// A rewritten version of `kettle.dataSource.CouchDB.read` that allows rewriting of the response using the
+// transformation rules found in `options.rules.read`.  We cannot directly extend the existing couch grade in Kettle
+// because it has a non-namespaced listener for `onRead`, which we cannot replace. The `gradeNames` and `listener` block
```

Comment:
We discussed this at the Eythorne summit of sixteen.  The two notations are equivalent.